### ISSUE TITLE
SWORD: add table of how dcterms maps to dataset fields

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#1077]
In DVN 3.6 we had "Dublin Core (DC) Qualified Mapping - DDI - Dataverse Network DB Element Crosswalk" at https://github.com/IQSS/dvn/blob/develop/doc/sphinx/source/dataverse-api-main.rst#dublin-core-dc-qualified-mapping---ddi---dataverse-network-db-element-crosswalk

We should probably add a similar table to the SWORD API Guide. @posixeleni said she'd do it. Thanks!

**Comment by posixeleni** [comment]
Completed this as a spreadsheet in Google drive: https://docs.google.com/spreadsheets/d/1r5jDqw9qMeeHaZ2ga8PSXTtom58gsmd5If3Skr9ercw/edit?usp=sharing Will add this to the documentation with the release of 4.0.

**Comment by eaquigley** [comment]
@posixeleni since this is ready, could you go ahead and add it in to documentation so we can get this ticket closed? thanks!

**Comment by posixeleni** [comment]
@eaquigley after meeting with Merce and Gustavo last week this ticket is actually not ready to be finished yet. I am waiting on this ticket #1430  to close before I can add this mapping to the documentation.

**Comment by posixeleni** [comment]
Assigning to @pdurbin to review

**Comment by posixeleni** [comment]
@axfelix @jwhitney I published the DC Terms to Dataverse DB fields Crosswalk for our 4.0 SWORD API documentation so if you have a moment to review that this all looks kosher from the OJS POV that would be fantastic!

Thanks!

Dublin Core Terms (DC Terms) Qualified Mapping - Dataverse DB Element Crosswalk
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|DC (terms: namespace)        |                Dataverse DB Element          |   Required   |                                                                     Note                                                                                    |
+=============================+==============================================+==============+=============================================================================================================================================================+
|dcterms:title                |                    title                     |       Y      |  Title of the Dataset.                                                                                                                                      |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:creator              |         authorName (LastName, FirstName)     |       Y      |  Author(s) for the Dataset.                                                                                                                                 |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:subject              |   subject (Controlled Vocabulary) OR keyword |       Y      |  Controlled Vocabulary list is in our User Guide > `Metadata References <http://guides.dataverse.org/en/latest/user/appendix.html#metadata-references>`_.   |                                                                                                                
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:description          |              dsDescriptionValue              |       Y      |  Describing the purpose, scope or nature of the Dataset. Can also use dcterms:abstract.                                                                     |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:publisher            |                 producerName                 |              |  Person or agency financially or administratively responsible for the Dataset                                                                               |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:contributor          |               datasetContactEmail            |       Y      |  Contact Email is required so will need to add an attribute type="Contact". Also used for Funder: add attribute type="Funder" which maps to contributorName.|                                                                                                                 
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:date                 |productionDate (YYYY-MM-DD or YYYY-MM or YYYY)|              |  Production date of Dataset.                                                                                                                                |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:type                 |                  kindOfData                  |              |  Type of data included in the file: survey data, census/enumeration data, aggregate data, clinical.                                                         |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:language             |                   language                   |              |  Language of the dataset (not the metadata). `Controlled Vocabulary list <http://guides.dataverse.org/en/latest/user/appendix.html#metadata-references>`_.  |                                                                                                                                               
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:identifier           |                 otherIdValue                 |              |  Other ID's that directly belong to dataset. If attribute="agency" is included map value to otherIdAgency (e.g., DOI, arXiv, handle, etc).                  |                              
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:source               |                 dataSources                  |              |  List of books, articles, data files if any that served as the sources for the Dataset.                                                                     |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:relation             |               relatedMaterial                |              |  Any related material (journal article citation is not included here - see: dcterms:isReferencedBy below).                                                  |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:coverage             |              otherGeographicCoverage         |              |  General information on the geographic coverage of the Dataset.                                                                                             |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:spatial              |              otherGeographicCoverage         |              |  General information on the geographic coverage of the Dataset.                                                                                             |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:license              |                   license                    |              |  Set the license to CC0 (default in Dataverse for new Datasets), otherwise enter "NONE" and fill in the dcterms:rights field.                               |                
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:rights               |                 termsofuse                   |              |  If not using CC0, enter any terms of use or restrictions for the Dataset.                                                                                  |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|dcterms:isReferencedBy       |             publicationCitation              |              |  The publication (journal article, book, other work) that uses this dataset (include citation, permanent identifier (DOI), and permanent URL).              |
+-----------------------------+----------------------------------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+

**Comment by axfelix** [comment]
Hey Eleni,

Thanks for this. Looks sensible with the exception of dcterms:coverage; I don't think that's supposed to be equated to dcterms:spatial, but I don't think it'll be a huge issue in many cases.

Also, your friendly neighbourhood pedant wants you to know that if you're formatting something like a table as plaintext it's nice not to have more than 80 characters per line :)

**Comment by pdurbin** [comment]
@axfelix a quick comment for you is that you can see a more readable HTML version at http://guides.dataverse.org/en/latest/api/sword.html

@posixeleni I'm giving this ticket back to you because I found a few issues. The first thing I did (in daa0559 in a branch) was re-order the sample XML file to put the fields in the same order as your crosswalk. If find this easier to follow and I like how you've put required fields at the top.

In the same commit I also added two fields to the sample XML that appears in your crosswalk but did not appear in the the sample XML and continue to not appear in the [SQL reference data](https://github.com/IQSS/dataverse/blob/daa0559243ac8073458b43f2a6749b95306ac74e/scripts/database/reference_data.sql):

- dcterms:language
- dcterms:spatial

Are we really supposed to support these fields with SWORD? They were never in the old crosswalk at http://guides.dataverse.org/en/3.6.2/dataverse-api-main.html#dublin-core-dc-qualified-mapping-ddi-dataverse-network-db-element-crosswalk (and I noticed dcterms:bibliographicCitation appears there in the 3.6.2 crosswalk... I'm not sure what the story is there.) If so, they must be added to the SQL reference data (and tested). As things stand right now these two fields are ignored by SWORD so we should either take them out of the crosswalk or add them to the SQL reference data and test that they work properly. In the screenshot below you won't see "English", which I used in dcterms:language nor "Boston" which I used in dcterms:spatial (I'm not sure if "Boston" is a good example value here). Again, they are ignored:

![roasting_at_home_-_trees_dataverse_-_2015-04-27_10 15 13](https://cloud.githubusercontent.com/assets/21006/7349567/144f0f10-ecc8-11e4-8711-a6e0e8769f89.png)

Those seem to be problems but I have other comments. dcterms:identifier is commented out and I'm not sure what the intended behavior is when it's used. Do you know? @scolapasta do you? We should enable the field and test and document its behavior. I'm seeing stuff about "If attribute='agency'" in the crosswalk but there's nothing in the SQL reference data to indicate than anything is done with that attribute.

Finally, the crosswalk table might be a nice place to add a column indicating if the value permits multiple values or not. I know we sort of indicate this in the XML example but it might be nice to state it explicitly.

**Comment by pdurbin** [comment]
@posixeleni as a followup to what I wrote above in 9707aea I removed the three fields from the XML example and your crosswalk that I feel have undefined behavior:

- dcterms:identifier
- dcterms:spatial
- dcterms:language
 
Let's work with @scolapasta on the plan for these fields since they require a change to the SQL reference data at least, possibly a code change, and testing.

**Comment by pdurbin** [comment]
@posixeleni and @scolapasta should we move this to QA? We'd like to tag 4.0 soon.

**Comment by posixeleni** [comment]
@pdurbin @scolapasta OK with me to send to QA. Created a new ticket to work on the items phil mentioned in https://github.com/IQSS/dataverse/issues/1077#issuecomment-96677687 and https://github.com/IQSS/dataverse/issues/1077#issuecomment-96728984

**Comment by pdurbin** [comment]
> OK with me to send to QA

Ok, I passed it to QA. Thanks for creating the new ticket (#2187) for future development, @posixeleni

**Comment by eaquigley** [comment]
Spoke with @posixeleni and got confirmation that this is good to go. Any new issues that may arise should be entered as a new issue. Closing ticket.
